app/controllers/registrations_controller.rb

```diff
@@ -378,6 +382,53 @@ def stripe_webhook
 
       stored_intent = stored_record.payment_intent
       stored_intent.update_status_and_charges(connected_account, stripe_intent, audit_event, audit_remote_timestamp)
+    when StripeWebhookEvent::REFUND_CREATED, StripeWebhookEvent::REFUND_UPDATED
+      # stripe_intent contains a Stripe::Refund as per Stripe documentation
+
+      original_charge = StripeRecord.charge.find_by(stripe_id: stripe_intent.charge)
+
+      if original_charge.nil?
+        # We created a record because refund creation is an incoming event,
+        #   but now we figured out that we don't care about it after all.
+        stored_record.destroy
+
+        logger.error "Stripe webhook reported a refund on charge #{stripe_intent.charge} but we never issued that one"
+        return head :not_found
+      elsif stored_record.parent_record.nil?
```

Comment:
Kinda redundant `if` (Rails is smart enough to skip redundant updates internally)